authentication/views.py

```diff
@@ -288,3 +291,75 @@ def get_user_details_for_keycloak(request, email):
             return Response(response, status=status.HTTP_200_OK)
     else:
         raise Http404("User not found")
+
+
+@api_view(["POST"])
+@permission_classes([IsAuthenticated])
+def post_request_password_update(request):
+    """
+    Endpoint for initiating the Keycloak password reset workflow via email.
+
+    Returns:
+        Response: HTTP_200_OK if the user and user's social auth record is found.
+
+    Raises:
+        Http404 if the user is not found.
+    """
+    user = getattr(request, "user", None)
+    if user:
+        strategy = load_strategy(request)
+        storage = strategy.storage
+        user_social_auth_record = storage.user.get_social_auth_for_user(
+            user, provider=OlOpenIdConnectAuth.name
+        ).first()
+        if user_social_auth_record:
+            payload = json.dumps(["UPDATE_PASSWORD"])
+            client_id = settings.SOCIAL_AUTH_OL_OIDC_KEY
+            keycloak_realm_name = settings.KEYCLOAK_REALM_NAME
+            keycloak_base_url = settings.KEYCLOAK_BASE_URL
+            url = f"{keycloak_base_url}/admin/realms/${keycloak_realm_name}/users/{user_social_auth_record.uid}/execute-actions-email?client_id={client_id}"
+            access_token = user_social_auth_record.get_access_token(strategy)
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": f"Bearer {access_token}",
+            }
+            response = requests.request(
+                "PUT",
+                url,
+                headers=headers,
+                data=payload,
+                timeout=settings.REQUESTS_TIMEOUT,
+            )
+            if response.status_code == HTTP_204_NO_CONTENT:
+                return Response({response}, status=status.HTTP_200_OK)
+    raise Http404("User not found")
+
+
+class CustomLogoutView(views.LogoutView):
+    """
+    Ends the user's Keycloak session in additional to the built in Django logout.
+    """
+
+    def post(self, request, *args, **kwargs):
+        """
+        POST endpoint for logging a user out.
+        Raises 404 if the user is not included in the request.
+        """
+        user = getattr(request, "user", None)
+        if user:
+            logout_of_keycloak(user)
```

Comment:
Logout seems to work, but I still get logged in automatically as the same user again when I click the sign-in link, no prompt for username/password